android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -1930,6 +2002,49 @@ public static void setGlobeKeyState(GlobeKeyState state) {
     globeKeyState = state;
   }
 
+  /**
+   * Handle the globe key action
+   * @param globeKeyDown boolean if the globe key state is GLOBE_KEY_STATE_DOWN
+   * @param keyboardType KeyboardType KEYBOARD_TYPE_INAPP or KEYBOARD_TYPE_SYSTEM
+   */
+  private static void handleGlobeKeyAction(Context context, boolean globeKeyDown, KeyboardType keyboardType) {
+    // Update globeKeyState
+    if (globeKeyState != GlobeKeyState.GLOBE_KEY_STATE_LONGPRESS) {
+      globeKeyState = globeKeyDown ? GlobeKeyState.GLOBE_KEY_STATE_DOWN : GlobeKeyState.GLOBE_KEY_STATE_UP;
+    }
+
+    if (KMManager.shouldAllowSetKeyboard()) {
+      KeyguardManager keyguardManager = (KeyguardManager) appContext.getSystemService(Context.KEYGUARD_SERVICE);
+      // inKeyguardRestrictedInputMode() deprecated, so check isKeyboardLocked to determine if screen is locked
+      if (keyguardManager.isKeyguardLocked()) {
+        if (globeKeyState == GlobeKeyState.GLOBE_KEY_STATE_UP) {
+          doGlobeKeyLockscreenAction(context);
+        }
+      } else {
+        // If screen isn't locked, reset the starting index
+        sysKbStartingIndexOnLockScreen = -1;
+
+        // Normal handling for globe key
+        if (globeKeyState == GlobeKeyState.GLOBE_KEY_STATE_LONGPRESS) {
+          // Longpress globe
+          doGlobeKeyLongpressAction(context, keyboardType);
+
+          // clear globeKeyState
+          globeKeyState = GlobeKeyState.GLOBE_KEY_STATE_UP;
+        } else if (globeKeyState == GlobeKeyState.GLOBE_KEY_STATE_UP) {
+          // Shortpress globe
+          doGlobeKeyShortpressAction(context, keyboardType);
+        }
+      }
+    } else {
+      // clear globeKeyState
+      globeKeyState = GlobeKeyState.GLOBE_KEY_STATE_UP;
+
+      // System Keyboard previously switched to next keyboard even though !shouldAllowSetKeyboard
+      //switchToNextKeyboard(context);
```

Comment:
The existing system keyboard code allows for this. Do we keep that?

https://github.com/keymanapp/keyman/blob/3f8d8f5cff56444ea291f68a8e12cf5c2ffc3317/android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java#L2352-L2357

Reply:
Hmm, perhaps we should split this change into a separate PR so we can more easily track it if problems arise?

Reply:
OK. I've enabled it with a TODO comment to address